_gsocproposals/2020/proposal_ATLAS-Visualization.md

```diff
@@ -0,0 +1,63 @@
+
+
+Thus, as part of the current software modernization effort in ATLAS, the replacement of the core 3D graphics engine is planned. The new [Qt3D toolkit](https://doc.qt.io/qt-5/qt3d-index.html), recently added to the official releases of the Qt framework, could be an excellent option. It would let developers easily integrate 3D graphics natively in the Qt GUI framework (already used in ATLAS for GUIs), *de-facto* removing the need of interface packages like SoQt.
+In addition to that, new features will be developed: to interact with the 3D objects composing the detector geometry more effectively; to let physicists see the details of the apparatus by the use of a better volumes' rendering and of animated views; to export 3D data to common formats.
+
+
+Development of new features will be performed in the context of the [new, standalone version](https://gitlab.cern.ch/GeoModelDev/geomodelvisualization) of the ATLAS visualization tools. That will let developers work outside the complex and specific software framework of the experiment, with a much faster development cycle. Moreover, that would leverage the experiment-independent character of the new tools, to potentially develop software solutions for the HEP community. In addition to that, new developments to the open source libraries used in the project could be eventually contributed back into the relevant code bases.  
+
+
+## Task ideas
+
+ * Develop a Qt3D testbed, to explore its possibilities
+ * Survey the other needed functionality (*e.g.*, the *SoSwitch* or *SoSelection* classes found in OpenInventor) and look for the existence of viable equivalents in Qt3D. (The graphics tools in VP1 is a possible example of what should be aimed for)
+ * Develop extended orbit controls
+ * Develop better volumes filtering
+ * Develop interactive picking of 3D geometry volumes
+ * Improve the handling of Boolean volumes, polycones, and tessellated solids
+ * Improve rendering of 3D volumes for detector geometry
+ * Improve the export of 3D data to common formats
+ * Animation and transitions between views of different parts of the detector geometry
+
+## Expected results
+
+The integration of a modern graphics engine, new software tools focusing on detector geometry, possibility to animate the views to better explore the volumes, the ability to export 3D data to common formats.
+
+## Requirements
+
+C++, Qt5 framework (3D graphics experience a bonus)
+
+
+## Mentors
```

Comment:
All the mentors have to be in mentors.md

Reply:
Done.